app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Api::V1::UsersController < ApplicationController
+  skip_before_action :authorized
+
+  def create
+    @user = User.create(user_params)
+    if @user.valid?
+      token = encode_token({ user_id: @user.id })
+      render json: { user: @user.username, token: token }, status: :ok
+    else
+      render json: { error: 'Invalid username' }, status: 400
+    end
+  end
+
+  def login
+    @user = User.find_by(username: params[:username])
```

Comment:
I would sanitize this. A malicious person can inject SQL scripts on the username field